crates/tbdex/src/http_client/mod.rs

```diff
@@ -71,3 +80,37 @@ fn generate_access_token(pfi_did_uri: &str, bearer_did: &BearerDid) -> Result<St
 
     Ok(access_token)
 }
+
+pub(crate) fn get_service_endpoint(pfi_did_uri: &str) -> Result<String> {
+    let resolution_result = ResolutionResult::new(pfi_did_uri);
+
+    let endpoint = match &resolution_result.document {
+        None => {
+            return Err(match resolution_result.resolution_metadata.error {
+                Some(e) => HttpClientError::Resolution(e),
+                None => HttpClientError::Resolution(ResolutionMetadataError::InternalError),
+            })
+        }
+        Some(d) => match &d.service {
+            None => {
+                return Err(HttpClientError::MissingServiceEndpoint(
+                    pfi_did_uri.to_string(),
+                ))
+            }
+            Some(s) => s
+                .iter()
+                .find(|s| s.r#type == *"PFI")
```

Comment:
should probably make PFI a parameter, 
an optional type parameter, if none specified it will return first or something